src/Core/src/Handlers/HybridWebView/HybridWebViewHandler.Windows.cs

```diff
@@ -104,30 +102,72 @@ private void OnWebMessageReceived(WebView2 sender, CoreWebView2WebMessageReceive
 
 		private async void OnWebResourceRequested(CoreWebView2 sender, CoreWebView2WebResourceRequestedEventArgs eventArgs)
 		{
-			// Get a deferral object so that WebView2 knows there's some async stuff going on. We call Complete() at the end of this method.
-			using var deferral = eventArgs.GetDeferral();
-
-			var (stream, contentType, statusCode, reason) = await GetResponseStreamAsync(eventArgs.Request.Uri);
-			var contentLength = stream?.Size ?? 0;
-			var headers =
-				$"""
-				Content-Type: {contentType}
-				Content-Length: {contentLength}
-				""";
-
-			eventArgs.Response = sender.Environment!.CreateWebResourceResponse(
-				Content: stream,
-				StatusCode: statusCode,
-				ReasonPhrase: reason,
-				Headers: headers);
-
-			// Notify WebView2 that the deferred (async) operation is complete and we set a response.
-			deferral.Complete();
+			var url = eventArgs.Request.Uri;
+
+			var logger = MauiContext?.CreateLogger<HybridWebViewHandler>();
+
+			logger?.LogDebug("Intercepting request for {Url}.", url);
+
+			// 1. First check if the app wants to modify or override the request.
+			{
+				// 1.a. First, create the event args
+				var platformArgs = new WebResourceRequestedEventArgs(sender, eventArgs);
+
+				// 1.b. Trigger the event for the app
+				var handled = VirtualView.WebResourceRequested(platformArgs);
+
+				// 1.c. If the app reported that it completed the request, then we do nothing more
+				if (handled)
+				{
+					logger?.LogDebug("Request for {Url} was handled by the user.", url);
+
+					return;
+				}
+			}
+
+			// 2. If this is an app request, then assume the request is for a local resource.
+			if (new Uri(url) is Uri uri && AppOriginUri.IsBaseOf(uri))
+			{
+				logger?.LogDebug("Request for {Url} will be handled by .NET MAUI.", url);
+
+				// 2.a. Get a deferral object so that WebView2 knows there's some async stuff going on. We call Complete() at the end of this method.
+				using var deferral = eventArgs.GetDeferral();
+
+				// 2.b. Check if the request is for a local resource
+				var (stream, contentType, statusCode, reason) = await GetResponseStreamAsync(eventArgs.Request.Uri, logger);
+
+				// 2.c. Create the response header
+				var headers = "";
```

Comment:
Can the request have headers we need to pass to the response?

Reply:
This is the internal/bundled files and cannot be intercepted.